lisa/variable.py

```diff
@@ -300,7 +300,13 @@ def add_secrets_from_pairs(
                 'secret variable, follow the formant "s:name:value"'
                 f"The raw error: {identifier}"
             )
-        _add_variable(key, value, results, is_secret=is_secret)
+        match = secret.PATTERN_URL[0].findall(str(value))
```

Comment:
as we discussed, it doesn't need to handle urls here. Does the variable section work?

Reply:
I changed add_secret as below:
```
def add_secret(
    origin: Any,
    mask: Optional[Union[Pattern[str], Tuple[Pattern[str], str]]] = None,
    sub: str = "******",
) -> None:
    global _secret_list
    if origin:
        if not isinstance(origin, str):
            origin = str(origin)
        if origin in _secret_set:
            for i in range(len(_secret_list)):
                if origin == _secret_list[i][0]:
                    _secret_list[i] = (origin, replace(origin, sub=sub, mask=mask))
                    break
        else:
            _secret_set.add(origin)
            _secret_list.append((origin, replace(origin, sub=sub, mask=mask)))
            # deal with longer first, in case it's broken by shorter
            _secret_list = sorted(_secret_list, reverse=True, key=lambda x: len(x[0]))
```
This works well. 
After calling ` __post_init__`  of `AzureNodeSchema`, the vhd url mask is replaced as the new one in _secret_list. I can see this when debugging.  

But after  ` __post_init__` , it no longer prints vhd parameter info. So we still couldn't see the vhd file name, but just "******".

Reply:
it's ok. The first priority is to protect the secret, secondly, it can show the original value help troubleshooting. Both requirements meet.

Reply:
OK, I posted a new update, please help to review again.